### PR TITLE
add board canzero

### PR DESCRIPTION
**Pull request by karu2003**
add board canzero small RPi board with CAN bus. https://github.com/ipmgroup/canzero-board_hardware

### CODE REVIEW

**Review by peterbarker — CHANGES_REQUESTED** [review]
@karu2003 the commit list needs cleaning up; should have a single commit per logical ArduPilot section - se "Preparing commits" here: https://ardupilot.org/dev/docs/submitting-patches-back-to-master.html

File: .github/workflows/test_linux_sbc.yml [hunk 1]
```diff
@@ -149,7 +149,8 @@ jobs:
             bebop,
             erlebrain2,
             pxfmini,
-            pxf
+            pxf,
+            canzero
```
Comment (peterbarker) [inline .github/workflows/test_linux_sbc.yml:153]:
```suggestion
            canzero,
```

... to simplify future diffs

**Review by peterbarker — CHANGES_REQUESTED** [review]

**Comment by magicrub** [comment]
@karu2003 instead of merge, please use rebase.
try ```git rebase ardupilot/master``` or ```git rebase origin/master```, whichever you have pointing to this repo

**Comment by karu2003** [comment]
Is there any hope that the PR will be accepted?

**Comment by lucasdemarchi** [comment]
@karu2003  why are there so many updates to this PR? Is this working on the HW and ready to be reviewed?

**Comment by karu2003** [comment]
> @karu2003 why are there so many updates to this PR? Is this working on the HW and ready to be reviewed?

This is working HW.  https://www.youtube.com/watch?v=7-TG8SYGsUo   https://www.youtube.com/watch?v=TGlhYWIwu8A

**Comment by magicrub** [comment]
You don't need to constantly rebase. When it merges it will be on top

**Comment by lucasdemarchi** [comment]
@karu2003  see https://github.com/ArduPilot/ardupilot/pull/25206#pullrequestreview-1691351416
You still need to split this commit to be acceptable and as @magicrub said, no need to keep rebasing it.

Let me know if you know how to do it, otherwise I can help.

**Comment by karu2003** [comment]
> @karu2003 see [#25206 (review)](https://github.com/ArduPilot/ardupilot/pull/25206#pullrequestreview-1691351416) You still need to split this commit to be acceptable and as @magicrub said, no need to keep rebasing it.
> 
> Let me know if you know how to do it, otherwise I can help.

Fine. I won't rebase and push in this branch. But. I made one commit as you asked. With all the proposed changes.

**Comment by lucasdemarchi** [comment]
@karu2003 just pushed here the same changes you had, but with separate commits per dir as required by Ardupilot. I also cleaned up some whitespaces that do not follow the coding style, but otherwise the diff is empty: https://github.com/ArduPilot/ardupilot/compare/ed4faf351f67d9a2a7740dc514a5ffba8f53d002..ff45b28f11b96efee4bbde28450d8d4889a19652

**Review by lucasdemarchi — APPROVED** [review]

**Comment by karu2003** [comment]
> @karu2003 just pushed here the same changes you had, but with separate commits per dir as required by Ardupilot. I also cleaned up some whitespaces that do not follow the coding style, but otherwise the diff is empty: https://github.com/ArduPilot/ardupilot/compare/ed4faf351f67d9a2a7740dc514a5ffba8f53d002..ff45b28f11b96efee4bbde28450d8d4889a19652

thank you more, I understood the ideology of your rules. I'll take this into account in the next RP.

**Comment by rmackay9** [comment]
By the way, recently we've had @Hwurzburg also approve (or at least review) our new board PR to AP.  The reason is that he nearly always turns up a problem as part of his effort to get new boards on the wiki.  This is Linux so not a blocker but you may want to add HenryW as a review in the future.

**Pull request merged**